### PR TITLE
Use intradoc-links and explain the various loading methods

### PR DESCRIPTION
**Pull request by MarijnS95**
Still had this in stash locally, might be nice for the next release.

### CODE REVIEW

File: ash/src/lib.rs [hunk 1]
```diff
@@ -25,6 +25,12 @@
 //! # Ok(()) }
 //! ```
 //!
+//! ## Getting started
+//! Load the Vulkan library at the default location using [`Entry::new()`][EntryCustom<_>::new()],
+//! or at a custom location using [`Entry::with_library("path/to/vulkan")`][EntryCustom<_>::with_library()].
```
Comment (MarijnS95) [inline ash/src/lib.rs:30]: Rustdoc doesn't seem capable of finding functions on type aliases yet :/

```rust
29 | //! Load the Vulkan library at the default location using [`Entry::new()`],
   |                                                            ^^^^^^^^^^^^^^ the type alias `Entry` has no function named `new`
```
So we're forced to specify `EntryCustom<_>::` directly.

File: ash/src/entry.rs [hunk 2]
```diff
@@ -168,8 +168,10 @@ impl<L> EntryCustom<L> {
         &self.entry_fn_1_1
     }
 
-    #[deprecated = "This function is unavailable and therefore panics on Vulkan 1.0, please use `try_enumerate_instance_version` instead"]
+    #[deprecated = "This function is unavailable and therefore panics on Vulkan 1.0"]
     #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkEnumerateInstanceVersion.html>"]
+    ///
+    /// Please use [`Self::try_enumerate_instance_version`] instead.
```
Comment (MarijnS95) [inline ash/src/entry.rs:174]: Intradoc links don't work in deprecation warnings - perhaps we should leave the original function name in the deprecation message?

File: ash/src/entry.rs [hunk 3]
```diff
@@ -10,7 +10,7 @@ use std::os::raw::c_char;
 use std::os::raw::c_void;
 use std::ptr;
 
-/// Function loader
+/// Hold `static` and `entry` functions
```
Comment (Ralith) [inline ash/src/entry.rs:13]:
```suggestion
/// Holds `static` and `entry` functions
```

I don't think "`static` and `entry`" is very helpful to someone who doesn't already understand this type. Maybe a bit more verbosity is justified?

File: ash/src/entry.rs (hunk 2 above)
Comment (Ralith) [reply ash/src/entry.rs:174]: I'd favor a helpful deprecation message over a link, especially since the link doesn't go very far in this case.

**Review by Ralith — APPROVED** [review]
Nice improvement!

File: ash/src/entry.rs (hunk 2 above)
Comment (MarijnS95) [reply ash/src/entry.rs]: Reverted the deprecation message, but left the link in docs in-place.

File: ash/src/entry.rs (hunk 3 above)
Comment (MarijnS95) [reply ash/src/entry.rs]: Added a bit more verbosity. Library users probably don't - should not - look at the type directly, but it's useful for those implementing their own loader. Couldn't find a proper definition of "entry" functions in Vulkan docs, but the table in `vkGetInstanceProcAddr` should be enough.

**Review by Ralith — APPROVED** [review]

**Pull request merged**